packages/aws-cdk/lib/api/toolkit-info.ts

```diff
@@ -73,6 +76,82 @@ export class ToolkitInfo {
     return { filename, key, changed: true };
   }
 
+  /**
+   * Prepare an ECR repository for uploading to using Docker
+   */
+  public async prepareEcrRepository(id: string, imageTag: string): Promise<EcrRepositoryInfo> {
+    const ecr = await this.props.sdk.ecr(this.props.environment, Mode.ForWriting);
+
+    // Create the repository if it doesn't exist yet
+    const repositoryName = 'cdk/' + id.replace(/[:/]/g, '-').toLowerCase();
```

Comment:
Does it really make sense to use a `cdk/` prefix? My intuition is that the repository name should be something the user controls, no?

Reply:
I'm treating this akin to the Asset bucket. Why should the user not be allowed to care about the asset bucket name, but be allowed to care about repository names?

Reply:
okay- I agree